frontend/src/components/Forms/MineRecordForm.js

```diff
@@ -65,39 +117,134 @@ export class MineRecordForm extends Component {
   removeField = (event, fields, index) => {
     event.preventDefault();
     fields.remove(index);
+    const removedMineType = this.props.mine_types[index].mine_tenure_type_code;
+    this.setState((prevState) => ({
```

Comment:
There's a lodash function `without` that might be handy here: https://lodash.com/docs/4.17.11#without
Should look something like...
```javascript
this.setState((prevState) => ({
  usedTenureTypes: without(prevState.usedTenureTypes, removedMineType)
}));
```

Reply:
done